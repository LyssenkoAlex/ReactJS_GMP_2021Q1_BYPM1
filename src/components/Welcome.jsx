import React from "react";
import Main from "./Main";
import Navigation from "./Navigation";
// eslint-disable-next-line import/no-named-as-default
import Search from "./header/Search";

const Welcome = () => (
  <>
    <Search />
    <Navigation />
    <Main />
  </>
);

export default Welcome;
