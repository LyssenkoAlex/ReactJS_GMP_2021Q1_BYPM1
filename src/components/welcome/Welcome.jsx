import React from 'react';
import Navigation from '../Navigation';
import Search from '../header/Search';

const Welcome = () => (
  <>
    <Search />
    <Navigation />
    <section>
      <h1>Start movie search</h1>
    </section>
  </>
);

export default Welcome;
