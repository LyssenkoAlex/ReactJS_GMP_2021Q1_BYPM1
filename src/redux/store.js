import { createStore, applyMiddleware } from "redux";
import rootReducer from "./reducers/reducers";
import apiMiddleware from "./middleware";

const store = createStore(rootReducer, applyMiddleware(apiMiddleware));
window.store = store;
export default store;
