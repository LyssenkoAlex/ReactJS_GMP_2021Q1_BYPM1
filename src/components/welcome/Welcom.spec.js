import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter} from 'react-router-dom';
import Welcome from './Welcome';
import store from '../../redux/store';

describe(`Test Welcome `, () => {
  it('renders Welcome', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Welcome />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
