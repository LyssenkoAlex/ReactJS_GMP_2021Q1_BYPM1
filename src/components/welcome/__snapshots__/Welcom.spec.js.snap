// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Welcome  renders Welcome 1`] = `
Array [
  <header>
    <div
      className="bcg_container"
    >
      <section
        className="upper_pane"
      >
        <section
          className="logo"
        >
          <span
            className="red_font red_font_bold"
          >
            netflix
          </span>
          <span
            className="red_font"
          >
            roulette
          </span>
        </section>
        <button
          className="add_movie_button"
          onClick={[Function]}
          role="add_movie"
          type="button"
        >
          Add Movie
        </button>
      </section>
      <section
        className="search_pane"
      >
        <h2>
          FIND YOUR MOVIE
        </h2>
        <div>
          <input
            onChange={[Function]}
            placeholder="What do you want to watch?"
            role="search_movie"
            type="search"
            value=""
          />
          <a
            href="/search_result/"
            onClick={[Function]}
          >
            <button
              onClick={[Function]}
              role="search"
              type="button"
            >
              Search
            </button>
          </a>
        </div>
      </section>
    </div>
  </header>,
  <nav>
    <ul
      className="menu"
    >
      <a
        href="/search_result"
        onClick={[Function]}
      >
        <li
          onClick={[Function]}
          value="All"
        >
          All
        </li>
      </a>
      <a
        href="/search_result"
        onClick={[Function]}
      >
        <li
          onClick={[Function]}
          value="Documentary"
        >
          Documentary
        </li>
      </a>
      <a
        href="/search_result"
        onClick={[Function]}
      >
        <li
          onClick={[Function]}
          value="Comedy"
        >
          Comedy
        </li>
      </a>
      <a
        href="/search_result"
        onClick={[Function]}
      >
        <li
          onClick={[Function]}
          value="Horror"
        >
          Horror
        </li>
      </a>
      <a
        href="/search_result"
        onClick={[Function]}
      >
        <li
          onClick={[Function]}
          value="Crime"
        >
          Crime
        </li>
      </a>
    </ul>
    <section
      className="sort_section"
    >
      <span
        className="sort_title"
      >
        SORT BY
      </span>
      <div
        className="dropdown-container"
      >
        <select
          className="select-css"
          onChange={[Function]}
        >
          <option
            value="genre"
          >
            genre
          </option>
          <option
            value="rating"
          >
            rating
          </option>
          <option
            value="release date"
          >
            release date
          </option>
        </select>
        <div
          className="select-icon"
        >
          <span>
            â–²
          </span>
        </div>
      </div>
    </section>
  </nav>,
  <section>
    <h1>
      Start movie search
    </h1>
  </section>,
]
`;
